ephypype/gather/gather_results.py

```diff
@@ -0,0 +1,266 @@
+
+                figures = glob.glob(op.join(fig_path, '*.png'))
+                if figures:
+                    for fig in figures:
+                        conn_figures.append(fig)
+
+                        data_path, basename, ext = split_filename(fig)
+
+                        caption = 'Subject: {} Session: {} Frequency band: {}'.format(sbj, ses, band)  # noqa
+                        captions.append(caption + ' epoch:' +
+                                        re.findall('\\d+', basename)[0])
+
+    return conn_figures, captions
+
+
+def get_psd_files(workflow_path, workflow_name, subject_id, session_id):
+    """
+    Get PSD file.
+
+    Parameters
+    ----------
+       workflow_path : str
+           Path of connectivity workflow
+       workflow_name : str
+           Name of the connectivity workflow
+       subject_id : list of str
+           List of subjects id
+       session_id : list of str
+           List of sessions
+
+    Returns
+    -------
+        psd_file : list of str
+            List of path of psd files
+        channel_coo_files : list of str
+            List of path of channel coordinates files
+    """
+    psd_pipeline = 'power_pipeline'
+    psd_node = 'power'
+
+    psd_files = list()
+    channel_coo_files = list()
+
+    subjects_folder = '_session_id_{ses}_subject_id_{sbj}'
+    node_path = op.join(workflow_path, workflow_name, psd_pipeline,
+                        subjects_folder, psd_node)
+
+    for sbj in subject_id:
+        for ses in session_id:
+            file_path = node_path.format(ses=ses, sbj=sbj)
+            psd_file = glob.glob(op.join(file_path, '*.npz'))
+            coo_file = glob.glob(op.join(file_path, '*coords.txt'))
+            if psd_file:
+                psd_files.append(psd_file[0])
+            if coo_file:
+                channel_coo_files.append(coo_file[0])
+
+    return psd_files, channel_coo_files
+
+
+def get_inverse_files(workflow_path, workflow_name, subject_id, session_id):
+    """
+    Get source reconstruction files.
+
+    Parameters
+    ----------
+       workflow_path : str
+           Path of connectivity workflow
+       workflow_name : str
+           Name of the connectivity workflow
+       subject_id : list of str
+           List of subjects id
+       session_id : list of str
+           List of sessions
+
+    Returns
+    -------
+        time_series_file : list of str
+            List of path of source reconstruction files
+        label_files : list of str
+            List of path of labels pickle file
+    """
+    inverse_pipeline = 'inv_sol_pipeline'
+    inverse_node = 'inv_solution'
+
+    time_series_files = list()
+    label_files = list()
+
+    subjects_folder = '_session_id_{ses}_subject_id_{sbj}'
+    node_path = op.join(workflow_path, workflow_name, inverse_pipeline,
+                        subjects_folder, inverse_node)
+
+    for sbj in subject_id:
+        for ses in session_id:
```

Comment:
why not use wildcard characters in glob so that you don't have to make a loop over sessions and subjects?